src/main/java/net/snowflake/ingest/utils/Utils.java

```diff
@@ -417,20 +417,21 @@ public static String getFullyQualifiedChannelName(
   }
 
   /**
-   * Get concat dot path, check if any path is empty or null
+   * Get concat dot path, check if any path is empty or null. Escape the dot field name to avoid
+   * column name collision.
    *
    * @param path the path
    */
   public static String concatDotPath(String... path) {
     StringBuilder sb = new StringBuilder();
     for (String p : path) {
-      if (isNullOrEmpty(p)) {
-        throw new IllegalArgumentException("Path cannot be null or empty");
+      if (p == null) {
+        throw new IllegalArgumentException("Path cannot be null");
       }
       if (sb.length() > 0) {
         sb.append(".");
       }
-      sb.append(p);
+      sb.append(p.replace("\\", "\\\\").replace(".", "\\."));
```

Comment:
does an empty string in p need to also be replaced by something? Have IT for an empty field name?

Reply:
I don't think empty string does not need to be replaced. An empty string does not collision with other dot path. An IT with empty fields was included in previous [PR](https://github.com/snowflakedb/snowflake-ingest-java/pull/869) in `IcebergStructuredIT.testFieldName`.